lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -588,29 +629,37 @@ def get_students_problem_grades(request, csv=False):
         module_state_key=module_state_key,
         module_type__in=PROB_TYPE_LIST,
         grade__isnull=False,
-    ).values('student__username', 'student__profile__name', 'grade', 'max_grade').order_by('student__profile__name')
+    ).values('student__id', 'student__username', 'student__profile__name', 'grade', 'max_grade').order_by('student__profile__name')
+
+    course_key = locator.CourseLocator.from_string(course_id)
+    non_student_id_list = get_non_student_list(course_key)
+    num_non_student = len(non_student_id_list)
 
     results = []
+    non_student_skipped = 0
     if not csv:
         # Restrict screen list length
-        # Adding 1 so can tell if list is larger than MAX_SCREEN_LIST_LENGTH
+        # Adding 1 + num_non_student so can tell if list is larger than MAX_SCREEN_LIST_LENGTH. non-students will be skipped.
         # without doing another select.
-        for student in students[0:MAX_SCREEN_LIST_LENGTH + 1]:
-            student_dict = {
-                'name': student['student__profile__name'],
-                'username': student['student__username'],
-                'grade': student['grade'],
-            }
-
-            student_dict['percent'] = 0
-            if student['max_grade'] > 0:
-                student_dict['percent'] = round(student['grade'] * 100 / student['max_grade'])
-            results.append(student_dict)
+        for student in students[0:MAX_SCREEN_LIST_LENGTH + 1 + num_non_student]:
+            this_user_id = student['student__id']
+            if this_user_id not in non_student_id_list:
```

Comment:
I think we could also eliminate the this_user_id variable here and use student['student__id'] directly.